src/components/sigin/SignIn.tsx

```diff
@@ -23,26 +25,71 @@ const SignIn = () => {
     console.log(token);
     // apply to form data
   };
+
+  const [authenticated, setAuthenticated] = useState<boolean | null>(null);
+
+  useEffect(() => {
+    const token = localStorage.getItem("access_token"); // Retrieve the token from your storage or context
+    if (token) {
+      api.auth.checkAuthentication(token).then((isAuthenticated) => {
+        setAuthenticated(isAuthenticated);
+      });
+    } else {
+      setAuthenticated(false);
+    }
+  }, []);
+
+  useEffect(() => {
+    if (authenticated) {
+      router.push("/");
+    }
+  }, [authenticated, router]);
+
+  const checkLogin = async (values: any) => {
+    const requestBody = JSON.stringify({});
+    const response = await axios.post(
+      API_URL_FROM_WEST + "/v1/auth/check",
+      requestBody,
+      {
+        headers: {
+          // Set your custom headers here
+          "Content-Type": "application/json",
+          Authorization: `Bearer ${values}`,
+        },
+      }
+    );
+
+    if (response.data.ok) {
+      console.log("can proceed to home now", response);
+      router.push("/");
+    }
+  };
   const handleSubmit = async (values: any) => {
     console.log("test", values);
     try {
-      const response = await fetch(API_URL+"/v1/auth/login", {
-        method: "POST",
-        "headers": {
-          "content-type": "application/json",
-        },
-        body: JSON.stringify({
-           ...values,
-           username: values.email
-        }),
+      const requestBody = JSON.stringify({
+        ...values,
+        username: values.email,
       });
-
-      console.log('response',response);
+      const response = await axios.post(
+        API_URL_FROM_WEST + "/v1/auth/login",
+        requestBody,
+        {
+          headers: {
+            // Set your custom headers here
+            "Content-Type": "application/json",
+          },
```

Comment:
Just fyi JSON.stringify() is not necessary at all. Just pass in the request body to axios and it'll handle it for you without adding extra headers as well.